indexing-solr/src/main/java/edu/unc/lib/boxc/indexing/solr/filter/SetDescriptiveMetadataFilter.java

```diff
@@ -445,6 +468,87 @@ private void extractDateCreated(Element mods, IndexDocumentBean idb) {
+
+        List<Element> rightsEls = mods.getChildren("accessCondition", JDOMNamespaceUtil.MODS_V3_NS);
+        for (Element rightsEl : rightsEls) {
+            String accessType = rightsEl.getAttributeValue("type");
+            if (accessType != null && accessType.trim().equalsIgnoreCase("use and reproduction")) {
+                String href = rightsEl.getAttributeValue("href", JDOMNamespaceUtil.XLINK_NS);
+                String modsRightsStatement = rightsEl.getTextTrim();
+                boolean hasUrl = !StringUtils.isBlank(href);
+                boolean hasRightsText = !StringUtils.isBlank(modsRightsStatement);
+
+                if (!hasRightsText && !hasUrl) {
+                    continue;
+                }
+
+                // No vocabulary. Add MODS text
+                if (hasRightsText && !hasUrl) {
+                    rights.add(modsRightsStatement);
+                }
+
+                if (hasUrl && !hasRightsText) {
+                    String trimmedHref = href.trim();
+                    if (rightsUriMap.containsKey(trimmedHref)) {
+                        String mappedRightsText = rightsUriMap.get(trimmedHref);
+                        rightsOaiPmh.add(setUrlProtocol(trimmedHref));
+                        rightsOaiPmh.add(mappedRightsText);
+                        rights.add(mappedRightsText);
+                    }
+                    rightsUri.add(trimmedHref);
+                }
+
+                if (hasRightsText && hasUrl) {
+                    String trimmedHref = href.trim();
+                    rightsUri.add(trimmedHref);
+                    rightsOaiPmh.add(setUrlProtocol(trimmedHref));
+
+                    String rightsMapText = rightsUriMap.get(trimmedHref);
+                    boolean hasRightsKey = rightsMapText != null;
+                    String indexingText = (hasRightsKey) ? rightsMapText : modsRightsStatement;
+                    rights.add(indexingText);
+                    rightsOaiPmh.add(indexingText);
+
+                    if (!hasRightsKey) {
+                        log.warn("URI, {} wasn't found in the rights uri mappings.", trimmedHref);
+                    } else if (!rightsMapText.equals(modsRightsStatement)) {
+                        log.warn("Rights text, {}, does not match the rights statement for, {}.",
+                                modsRightsStatement, href);
+                    }
+                }
+            }
+        }
+
+        if (!rights.isEmpty()) {
+            idb.setRights(rights);
+        } else {
+            idb.setRights(null);
+        }
+
+        if (!rightsOaiPmh.isEmpty()) {
+            idb.setRightsOaiPmh(rightsOaiPmh);
+        } else {
+            idb.setRightsOaiPmh(null);
+        }
+
+        if (!rightsUri.isEmpty()) {
+            idb.setRightsUri(rightsUri);
+        } else {
+            idb.setRightsUri(null);
+        }
+    }
+
+    private String setUrlProtocol(String uri) {
```

Comment:
since I usually try to avoid `set` for non-setter methods, something like `convertUrlProtocol` might be better